apps/wallet/src/ui/app/pages/swap/SwapPageForm.tsx

```diff
@@ -0,0 +1,460 @@
+import { MaxSlippage } from '_pages/swap/MaxSlippage';
+import { QuoteAssets } from '_pages/swap/QuoteAssets';
+import { FEES_PERCENTAGE, initialValues, type FormValues } from '_pages/swap/utils';
+import { validate } from '_pages/swap/validation';
+import { useCoinMetadata, useFormatCoin } from '@mysten/core';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
+import { ArrowDown12, ArrowRight16, ChevronDown16, Refresh16 } from '@mysten/icons';
+import { MIST_PER_SUI } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import clsx from 'classnames';
+import { Form, Formik, useFormikContext } from 'formik';
+import { useMemo, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+function useCoinTypeData(activeCoinType: string | null) {
+	const selectedAddress = useActiveAddress();
+
+	const { staleTime, refetchInterval } = useCoinsReFetchingConfig();
+
+	const { data: coins, isLoading: coinsLoading } = useSuiClientQuery(
+		'getAllBalances',
+		{ owner: selectedAddress! },
+		{
+			enabled: !!selectedAddress,
+			refetchInterval,
+			staleTime,
+			select: filterAndSortTokenBalances,
+		},
+	);
+
+	const activeCoin = coins?.find(({ coinType }) => coinType === activeCoinType);
+	const activeCoinBalance = activeCoin?.totalBalance;
+	const [tokenBalance] = useFormatCoin(activeCoinBalance, activeCoinType);
+	const coinMetadata = useCoinMetadata(activeCoinType);
+
+	return {
+		activeCoin,
+		tokenBalance,
+		coinMetadata,
+		isLoading: coinsLoading || coinMetadata.isLoading,
+	};
+}
+
+function SuiToUSD({ amount, isPayAll }: { amount: string; isPayAll: boolean }) {
+	const amountAsBigInt = new BigNumber(amount);
+	const { rawValue } = useSuiBalanceInUSDC(amountAsBigInt);
+
+	return (
+		<div className="text-bodySmall font-medium text-hero-darkest/40">
+			{isPayAll ? '~ ' : ''}
+			{getUSDCurrency(rawValue)}
+		</div>
+	);
+}
+
+function AssetData({
+	tokenBalance,
+	coinType,
+	symbol,
+	to,
+	onClick,
+	disabled,
+}: {
+	tokenBalance: string;
+	coinType: string;
+	symbol: string;
+	to?: string;
+	onClick?: () => void;
+	disabled?: boolean;
+}) {
+	return (
+		<DescriptionItem
+			title={
+				<div className="flex gap-1 items-center">
+					<CoinIcon coinType={coinType} size="sm" />
+					<ButtonOrLink
+						disabled={disabled}
+						onClick={onClick}
+						to={to}
+						className={clsx(
+							'flex gap-1 items-center no-underline outline-none border-transparent bg-transparent p-0',
+							!disabled && 'cursor-pointer',
+						)}
+					>
+						<Heading variant="heading6" weight="semibold" color="hero-dark">
+							{symbol}
+						</Heading>
+						{!disabled && <ChevronDown16 className="h-4 w-4 text-hero-dark" />}
+					</ButtonOrLink>
+				</div>
+			}
+		>
+			{!!tokenBalance && (
+				<div className="text-bodySmall font-medium text-hero-darkest/40">
+					{tokenBalance} {symbol}
+				</div>
+			)}
+		</DescriptionItem>
+	);
+}
+
+function getCoinFromSymbol(symbol: string) {
```

Comment:
Using Symbol is not safe I think. Anyone can create coins with SUI as symbol. It should be safer to use the coin type

Reply:
removed